crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -32,8 +35,12 @@ fn storage_key(address: &str) -> String {
     format!("{ACCOUNTS_PREFIX}/{address}")
 }
 
-pub(crate) fn balance_storage_key(address: Address) -> String {
-    format!("{}/balance", storage_key(&address.encode_hex::<String>()))
+pub(crate) fn balance_storage_key(address: Address, asset: &asset::Id) -> String {
+    format!(
+        "{}/balance/{}",
+        storage_key(&address.encode_hex::<String>()),
+        asset.as_bytes().encode_hex::<String>()
```

Comment:
See my comment above: `hex::ToHex` is implemented for all `T: AsRef<[u8]>`, so you could just `impl AsRef<[u8]> for Id` to make this a bit more elegant